lockfile_generator/src/lib.rs

```diff
@@ -157,12 +158,12 @@ impl Display for Error {
             Self::Io(_) => write!(f, "I/O error"),
             Self::Json(_) => write!(f, "json parsing error"),
             Self::NonZeroExit(output) => {
-                write!(
-                    f,
-                    "package manager quit unexpectedly (code: {:?}, signal: {:?})",
-                    output.status.code(),
-                    output.status.signal()
-                )?;
+                write!(f, "package manager quit unexpectedly (code: {:?}", output.status.code())?;
+                #[cfg(unix)]
+                {
+                    write!(f, ", signal: {:?}", output.status.signal())?;
+                }
```

Comment:
I'd assume the dedicated block here is necessary because of the way the macro is built? Usually a single statement that isn't an assignment can have cfg directly attached I thought.

Also this is likely cleaner by just printing `None` on Windows. I doubt anyone would mind.

Reply:
I've removed the block. I'm not sure how to print `None` on windows with making the code more complicated... So I've left that bit out.